README.md

```diff
@@ -22,10 +38,36 @@ $ composer require jmikola/auto-login-bundle=~1.0
 
 This bundle requires Symfony 2.1 or above. There is no support for Symfony 2.0.
 
-## Configuration
+## Usage and configuration
+
+You need to create a new `UserProvider` service that implements `Jmikola\AutoLogin\User\AutoLoginUserProviderInterface`. This 
+service is responsable to fetch the correct user object from the URL token. 
+
+This bundle registers a firewall listener, which is configured via the `jmikola_auto_login` key in your security 
+component's firewall configuration. See this example configuration: 
+
+```yml
+// services.yml
+services:
+  acme.auto_login_user_provider:
+    # Implements Jmikola\AutoLogin\User\AutoLoginUserProviderInterface
+    class: Acme\UserBundle\Security\AcmeAutoLoginUserProvider
+```
+```yml
+// security.yml
+security:
+  firewalls:
+    main:
+      # We need not specify a "provider" for our firewall or listeners,
+      # since SecurityBundle will default to the first provider defined.
+      jmikola_auto_login:
+        auto_login_user_provider: acme.auto_login_user_provider
+```
+
+In the example above we need to specify a custom service for `auto_login_user_provider`, since default
+`EntityUserProvider` does not implement `AutoLoginUserProviderInterface`.
 
-This bundle registers a firewall listener, which is configured via the
-`jmikola_auto_login` key in your security component's firewall configuration.
+When you go to the url `http://your-app.com/whatever?_al=foobar` we will invoke the `AcmeAutoLoginUserProvider::loadUserByAutoLoginToken` with parameter `foobar`. It is your job to make sure the `AcmeAutoLoginUserProvider` returns the correct user for that token. 
```

Comment:
"When visiting `http://example.com/path?_al=foobar`, we will invoke `AcmeAutoLoginUserProvider::loadUserByAutoLoginToken` with the value of `foobar`. You must ensure that `AcmeAutoLoginUserProvider` returns the correct user for that token (or throws accordingly)."